kong/templates/kong_defaults.lua

```diff
@@ -15,8 +15,9 @@ host_ports = NONE
 anonymous_reports = on
 
 proxy_listen = 0.0.0.0:8000 reuseport backlog=16384, 0.0.0.0:8443 http2 ssl reuseport backlog=16384
-stream_listen = off
 admin_listen = 127.0.0.1:8001 reuseport backlog=16384, 127.0.0.1:8444 http2 ssl reuseport backlog=16384
+stream_listen = off
+stream_api = NONE
```

Comment:
Do we have to make this configurable? To me it seems that this HTTP layer in stream is only for Kong internal use, which in that case a unix domain socket under the Kong prefix using a well known name should be enough.

Reply:
definitely unix domain socket is much better (and changed the tests to be a better example).
I'd also like to avoid a new config variable, but not sure if it's worth it to have an unconditional `listen`.  if was an internal not-really-a-socket, it wouldn't be an issue, but i don't think that exists for streams.